packages/block-library/src/navigation-link/edit.js

```diff
@@ -80,57 +45,23 @@ function NavigationLinkEdit( {
 	insertLinkBlock,
 } ) {
 	const { label, opensInNewTab, title, url, nofollow, description } = attributes;
-	const link = title ? { title: unescape( title ), url } : null;
-	const [ isLinkOpen, setIsLinkOpen ] = useState( ! label && isSelected );
-
-	let onCloseTimerId = null;
+	const link = {
+		title: title ? unescape( title ) : '',
+		url,
+		newTab: opensInNewTab,
+	};
+	const [ isLinkOpen, setIsLinkOpen ] = useState( ! url && isSelected );
 
 	/**
-	 * It's a kind of hack to handle closing the LinkControl popover
-	 * clicking on the ToolbarButton link.
+	 * This hack shouldn't be necessary but due to a focus loss happening
+	 * when selecting a suggestion in the link popover, we force close on block unselection.
```

Comment:
@aduth @gziolo not sure if you have an idea about this. Why there's a focus loss when we select a suggestion causing the `onClose`  of the Popover to never be called if we click outside.

Reply:
```js
const [ isLinkOpen, setIsLinkOpen ] = useState( ! url && isSelected );
```

It feels like this effect exists only because `isSelected` is checked only on the initial render. Did you try to refactor code to use `isLinkOpen && isSelected` as a guard clause for the link popover?

Reply:
It seems to work in my testing, but I might miss something:

```js
{ isLinkOpen && isSelected && <LinkControl  ... }
```

Reply:
yes, this works but I think we should fix the root issue instead which is the focus loss.

Reply:
I'll update the code to follow this suggestion in the meantime

Reply:
I actually think we should keep it (after testing) otherwise when we focus back the link will stay open.